File: smart_home_app/smart_home_core/lib/src/models/device_type.dart
enum DeviceType {
  airQuality,
  smartLeds,
  unknown;

  static DeviceType parse(String type) {
    switch (type) {
      case 'air_quality':
        return DeviceType.airQuality;
      case 'smart_leds':
        return DeviceType.smartLeds;
      default:
        return DeviceType.unknown;
    }
  }

  @override
  String toString() {
    switch (this) {
      case airQuality:
        return 'air_quality';
      case smartLeds:
        return 'smart_leds';
      default:
        return 'unknown';
    }
  }
}

File: smart_home_backend/src/main/java/jmojzes21/smart_home_backend/models/DeviceType.java
package jmojzes21.smart_home_backend.models;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "device_type", schema = "public")
public class DeviceType {

  public static final String SMART_LEDS = "smart_leds";
  public static final String AIR_QUALITY = "air_quality";

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  private Integer id;

  @Column(name = "key")
  private String key;

  public DeviceType() {}

  public Integer getId() {
    return id;
  }

  public void setId(Integer id) {
    this.id = id;
  }

  public String getKey() {
    return key;
  }

  public void setKey(String key) {
    this.key = key;
  }
}
